src/transport.js

```diff
@@ -67,7 +68,7 @@ module.exports = function (swarm) {
               log('dial canceled: %s', multiaddr.toString())
               // clean up already done dials
               if (conn) {
-                conn.close()
+                pull(pull.empty(), conn)
```

Comment:
What I'm thinking now is:

Connections should have a conn.close that sends a `pull.empty()` by default or call their parent instance version of conn.close. Transports should overload that method to do a destroy/close, whatever the transport prefers.

When a stream muxer is applied, we need to make sure that conn.close gets overloaded back to the conn.close that defaults to pull.empty, because closing a muxed stream should not destroy the socket not the muxer.

This is similar to what we do with getPeerInfo